dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataOwnersManagerTest.kt

```diff
@@ -76,4 +92,58 @@ class DataOwnersManagerTest {
         }
         assertTrue(exception.summary.contains("User is already a data owner for company."))
     }
+
+    @Test
+    fun `check that email for users becoming company data owner is not generated if company does not exist`() {
+        val mockStoredCompany = mock(StoredCompanyEntity::class.java)
+
+        `when`(mockStoredCompany.companyName).thenReturn(testCompanyName)
+        `when`(mockStoredCompany.companyId).thenReturn(UUID.randomUUID().toString())
+
+        val exception = assertThrows<ResourceNotFoundApiException> {
+            dataOwnersManager.addDataOwnerToCompany(
+                companyId = mockStoredCompany.companyId,
+                userId = testUserId,
+                companyName = mockStoredCompany.companyName,
+            )
+        }
+        assertTrue(exception.summary.contains("Company is invalid"))
+    }
+
+    @Test
+    fun `check that email generated for users becoming company data owner are generated`() {
+        val mockStoredCompany = mock(StoredCompanyEntity::class.java)
+        val companyId = UUID.randomUUID().toString()
+
+        `when`(mockStoredCompany.companyName).thenReturn(testCompanyName)
+        `when`(mockStoredCompany.companyId).thenReturn(companyId)
+        `when`(mockCompanyRepository.existsById(mockStoredCompany.companyId)).thenReturn(true)
+        `when`(mockDataOwnersRepository.existsById(mockStoredCompany.companyId)).thenReturn(false)
+        `when`(
+            dataOwnershipSuccessfullyEmailMessageSender.getNumberOfOpenDataRequestsForCompany(
+                mockStoredCompany.companyId,
+            ),
+        ).thenReturn(5)
+        `when`(dataOwnershipSuccessfullyEmailMessageSender.getEmailAddressDataOwner(testUserId))
+            .thenReturn("test@test.com")
```

Comment:
change email to: test@example.com (we always want to use ...@example.com because our email service doesnt sent emails to mails ending in @example.com)

Reply:
Thanks for the advice. It is modified.